gcc/rust/lex/rust-token.cc

```diff
@@ -172,6 +173,54 @@ Token::get_str () const
   return *str;
 }
 
+static std::string
+escape_special_chars (const std::string &source, bool in_string = false)
+{
+  std::stringstream ss;
+  for (char c : source)
+    {
+      switch (c)
+	{
+	case '\t':
+	  ss << "\\t";
+	  break;
+	case '\n':
+	  ss << "\\n";
+	  break;
+	case '\r':
+	  ss << "\\r";
+	  break;
+	case '\0':
+	  ss << "\\0";
+	  break;
+	case '\\':
+	  ss << "\\\\";
+	  break;
+	case '\v':
+	  ss << "\\v";
+	  break;
+	case '\'':
+	  if (in_string)
+	    ss << c;
+	  else
+	    ss << "\\'";
+	  break;
+	case '"':
+	  if (in_string)
+	    ss << "\\\"";
+	  else
+	    ss << c;
+	  break;
+	// Should we also replace unicode escape or \x digits ?
```

Comment:
@tamaroning what do you think?

Reply:
I think we should also convert characters back into unicode escapes and byte literal (\x).
If string literal contains other newlines than \n, dump file is broken. But it requires utf-8 decoding.